files/usr/share/jeos-firstboot/jeos-firstboot-functions

```diff
@@ -0,0 +1,136 @@
+	# stty size can return zero when not ready or
+	# its a serial console
+	if [ "$COLUMNS" = "0" -o "$LINES" = "0" ]; then
+		LINES=24
+		COLUMNS=80
+	fi
+}
+stty_size
+
+result=
+list=
+password=''
+
+let dh_menu=LINES-15
+let dh_text=LINES-5
+
+call_module_config() {
+        local module="$1"
+        module_function="${module}_jeos_config"
+        [ "$(type -t "${module_function}")" = "function" ] || return 1
+        "${module_function}" && true # To not trigger errexit
+        ret=$?
+        [ $ret -eq 0 ] || return $ret
+}
+
+kmscon_available() {
+	# kmscon itself is installed
+	kmscon --help >/dev/null 2>&1 || return 1
+	# At least one monospace font is available
+	[ -n "$(fc-match "monospace" 2>/dev/null)" ] || return 1
+
+	return 0
+}
+
+fbiterm_available() {
+	# fbiterm itself is installed
+	fbiterm --help >/dev/null 2>&1 || return 1
+	# fbiterm comes with its own fallback font
+
+	return 0
+}
+
+dialog_out=`mktemp -qt 'firstboot-XXXXXX'`
+d(){
+	retval=
+	while true
+	do
+		retval=0
+		dialog --backtitle "$PRETTY_NAME" --output-fd 3 "$@" 3>"${dialog_out}" || retval=$?
+		case $retval in
+		  0)
+			# need || true as dialog doesn't write newlines
+			read result < $dialog_out || true
+			return 0
+			;;
+		  1)
+			echo "$(xargs -a "$dialog_out")" >/var/log/jeos
+			dialog --yesno $"Do you really want to quit?" 0 0 && exit 1
+			continue
+			;;
+		  255)
+			# xargs to remove whitespaces
+			echo "$(xargs -a "$dialog_out")" >/var/log/jeos
+			result_error="$(xargs -a "$dialog_out")"
+			if [ -z "$result_error" ]; then
+				dialog --yesno $"Do you really want to quit?" 0 0 && exit 1
+				continue
+			fi
+			logger -p err -t jeos-firstboot "$result_error"
+			dialog --msgbox $"Exiting due to error, please check the system log" 0 0
+			exit 2
+			;;
+		esac
+	done
+}
+
+warn(){
+	d --title $"Warning" --msgbox "$1" 6 40
+}
+
+# localectl --no-pager list-keymaps does not list aliases (symlinks), but those are used
+# by YaST/langset.sh, so we need to show them.
+findkeymaps()
+{
+        list=()
+        local line
+        while read line; do
+                list+=("${line%.map.gz}" '')
+        done < <(find /usr/share/kbd/keymaps -name '*.map.gz' -printf "%f\n" | sort -u)
+        [ -n "$list" ]
+}
+
+findlocales()
+{
+        list=()
+        local l locale
+        # List only locales which are both in live-langset-data and glibc-locale(-base)
+        for l in /usr/share/langset/*; do
+                locale="${l#/usr/share/langset/}"
+                [ -d "/usr/lib/locale/${locale}.utf8" ] || continue
+                list+=("${locale}" '')
+        done
+        [ -n "$list" ]
+}
+
+get_current_locale()
+{
+	JEOS_LOCALE=`awk -F= '$1 == "LANG" { print $2; exit }' /etc/locale.conf`
+}
+
+apply_locale()
+{
+	if [ ! -z "$JEOS_LOCALE" ]; then
+		run langset.sh $JEOS_LOCALE || warn $"Setting the locale failed"
+	fi
+}
+
+apply_locale_and_keytable()
+{
+	if [ ! -z "$JEOS_LOCALE" -a ! -z "$JEOS_KEYTABLE" ]; then
```

Comment:
locale and keytable should be given as parameters

Reply:
This would mean change everything to parameters, now almost all functions are receiving data as global, to ease the work of loading the preset configurations, will put in the list of desirable things in Trello, but will leave like that for now.

Reply:
Fortunately it doesn't need many changes, just replace calls to `apply_locale_and_keytable` with `apply_locale_and_keytable "$JEOS_LOCALE" "$JEOS_KEYTABLE"` and it's much clearer, but with unchanged behaviour.

Reply:
alright, will take a look at this

Reply:
This I will leave to a separate review if you don't mind, this one is already too convoluted and I got politely asked to make smaller changes so it's easier for other people to look at.
As you know I'm taking notes about this smaller details so we can quickly create another PR with this clean-ups.